docs/pages/machine-workload-identity/machine-id/access-guides/mcp.mdx

```diff
@@ -0,0 +1,207 @@
+---
+title: Machine & Workload Identity with MCP Access
+description: How to use Machine & Workload Identity to access MCP servers
+labels:
+ - how-to
+ - mwi
+---
+
+Teleport protects and control access to Model Context Protocol (MCP) servers.
+Machine & Workload Identity (MWI) can then be used to grant machines and
+workloads secure and short-lived access to these MCP servers without the need
+for long-lived static secrets.
+
+In this guide, you will configure `tbot` to produce short-lived credentials that
+can be used by an MCP client to access MCP servers enrolled within your Teleport
+cluster.
+
+This guide focuses on granting machines access to MCP servers. If you wish to
+grant human users access to MCP servers, you should instead refer to the
+[MCP Access with Stdio MCP Server guide](../../../enroll-resources/mcp-access/stdio.mdx).
+
+## How it works
+
+The Teleport Application Access agent is deployed in front of the MCP server
+that you wish to protect access to. This agent is responsible for enforcing
+access control based on roles configured in Teleport.
+
+The Machine & Workload Identity agent, `tbot`, is installed on the machine
+that will require access to the MCP server. It is responsible for authenticating
+to the Teleport cluster and producing an identity file that can be used by the
+MCP client to access the MCP server through the Teleport Proxy.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- If you have not already connected your MCP server to Teleport, follow
+the [MCP Getting Started guide](../../../enroll-resources/mcp-access/getting-started.mdx).
+- (!docs/pages/includes/tctl.mdx!)
+- `tbot` and `tsh` must already be installed and configured on the machine that
+will access MCP servers. For more information, see the
+[deployment guides](../deployment/deployment.mdx).
+
+## Step 1/3. Configure RBAC
+
+First, Teleport must be configured to allow the credentials produced by the bot
+to access MCP servers in your infrastructure. In this example, access will be
+granted to all MCP servers and all tools within those servers. This is done by
+creating a role that grants the necessary permissions and then assigning this
+role to the Bot.
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: example-role
+spec:
+  allow:
+    app_labels:
+      '*': '*'
```

Comment:
I would not use wildcard app labels even as an example, it's a known footgun. E.g. if you have Okta integration setup and assign someone this role, they will get assigned to all Okta apps.